### PR TITLE
OCM-5461 | feat: use default version

### PR DESCRIPTION
**Pull request by chenz4027**
https://issues.redhat.com/browse/OCM-5461

### CODE REVIEW

File: cmd/create/machinepool/nodepool.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		clusterVersion := cluster.Version().RawID()
 		// This is called in HyperShift, but we don't want to exclude version which are HCP disabled for node pools
 		// so we pass the relative parameter as false
-		versionList, err := versions.GetVersionList(r, channelGroup, true, true, false, false)
+		_, versionList, err := versions.GetVersionList(r, channelGroup, true, true, false, false)
```
Comment (chenz4027) [inline cmd/create/machinepool/nodepool.go:75]: Should we also use the defaultVersion [here](https://github.com/openshift/rosa/blob/b6ca8f1112885cfb56cab8f4ba3a1a734005c195/cmd/create/machinepool/nodepool.go#L103)?

File: pkg/helper/versions/helpers.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ const (
 )
 
 func GetVersionList(r *rosa.Runtime, channelGroup string, isSTS bool, isHostedCP bool, filterHostedCP bool,
```
Comment (robpblake) [inline pkg/helper/versions/helpers.go:20]: @chenz4027 You're currently not meeting our requirements on adding test coverage for changes to ROSA which is why the build on your PR is failing.

From your changes, this looks like the easiest function to write a test for. If you can do that, your build should pass.

**Review by robpblake — CHANGES_REQUESTED** [review]

File: pkg/helper/versions/helpers.go (hunk 2 above)
Comment (chenz4027) [reply pkg/helper/versions/helpers.go:20]: Thanks @robpblake , I will add a test for it 👍

File: pkg/helper/versions/helpers.go [hunk 3]
```diff
@@ -32,13 +32,23 @@ func GetVersionList(r *rosa.Runtime, channelGroup string, isSTS bool, isHostedCP
...
+		if isHostedCP {
+			if v.HostedControlPlaneDefault() {
+				defaultVersion = v.RawID()
+			}
+
+		} else {
+			if v.Default() {
+				defaultVersion = v.RawID()
+			}
+		}
```
Comment (mikeshng) [inline pkg/helper/versions/helpers.go:44]:
```
		if (isHostedCP && v.HostedControlPlaneDefault()) || v.Default() {
				defaultVersion = v.RawID()
		}
```

Minior nit. Is this logically equivalent to the if block above?

File: pkg/ocm/versions.go [hunk 4]
```diff
@@ -87,6 +87,9 @@ func (c *Client) GetVersionsWithProduct(product string, channelGroup string,
 			request.Parameter("product", product)
 		}
 		response, err = request.Send()
+		// log response
+		fmt.Println("here!", response.Items())
```
Comment (mikeshng) [inline pkg/ocm/versions.go:91]: Just a reminder to remove this.

Comment (chenz4027) [reply pkg/ocm/versions.go]: Removed 👍

File: pkg/helper/versions/helpers.go (hunk 3 above)
Comment (chenz4027) [reply pkg/helper/versions/helpers.go]: Yep I moved it to a function

File: pkg/helper/versions/helpers.go [hunk 5]
```diff
@@ -55,6 +56,13 @@ func GetVersionList(r *rosa.Runtime, channelGroup string, isSTS bool, isHostedCP
 	return
 }
 
+func GetDefaultVersion(version *v1.Version, isHostedCP bool) string {
```
Comment (mikeshng) [inline pkg/helper/versions/helpers.go:59]:
```suggestion
func getDefaultVersion(version *v1.Version, isHostedCP bool) string {
```

If it doesn't need to be exposed then let's make it a private helper function.

File: cmd/create/machinepool/nodepool.go (hunk 1 above)
Comment (andreadecorte) [reply cmd/create/machinepool/nodepool.go:75]: for node pool default version is the cluster version (aka control plane). So no need to change it.

File: pkg/helper/versions/helpers.go (hunk 5 above)
Comment (mikeshng) [reply pkg/helper/versions/helpers.go]: @chenz4027 please feel free to resolve. You have addressed my comment. Thanks.

File: pkg/ocm/versions.go [hunk 6]
```diff
@@ -275,10 +275,6 @@ func GetVersionMinorList(ocmClient *Client) (versionList []string, err error) {
 	return
 }
 
-func (c *Client) GetDefaultVersion(channelGroup string) (version string, err error) {
```
Comment (andreadecorte) [inline pkg/ocm/versions.go:278]: unused?

Comment (chenz4027) [reply pkg/ocm/versions.go:278]: Yep

**Comment by andreadecorte** [comment]
lgtm

**Pull request merged**